site/docs/configuration.md

```diff
@@ -223,6 +223,45 @@ Preconstruct will enforce that the `exports` field that is written can is direct
 }
 ```
 
+### `type` (experimental) {#type}
+
+```ts
+"module" | "commonjs";
+```
+
+> If you're just thinking "I want to write code with the native ECMAScript import and export syntax and have my code work for most people", you likely should not use this feature. This feature allows you to build packages that import Node.js ESM only dependencies and therefore only support being imported from ESM in newer versions of Node.js and some bundlers.
+
+The `type` config being set to `"module"` allows you to generate ECMAScript modules(ESM) compatible with Node.js' implementation of ESM (Node.js ESM).
+
+You should likely only use this feature if you fit into one these categories:
+
+- You're interested in the state of modules in the JS ecosystem
+- You want to publish a package that synchronously imports a Node.js ESM only dependency
```

Comment:
"synchronously" might be a little bit misleading since it's not technically correct, imports are always fetched asynchronously - it's the initialization of the importing module that is deferred

Reply:
```suggestion
- You want to publish a package that statically imports a Node.js ESM only dependency
```